### DISCUSSION TITLE
Polymorphisms & Relationships Fields in the order I want

### DISCUSSION THREAD

**Discussion by antoinemy** [filamentphp/filament#5561] [Help] [answered]
Hello,

I have a `Product` table which has a `ProductCrafted` child polymorph. So I display the `ProductCrafted` data in my form and I access the parent data of my `Product` table through the `relationship()` function as in the example below:

```
Grid::make()
  ->relationship('product')
  ->schema([
      TextInput::make('name'),
      FileUpload::make('image'),
  ]),
TextInput::make('others_informations'),
TextInput::make('others_informations'),
```

However the `relationship()` function only applies to the `Grid`, `Fieldset` classes... So it's impossible for me to get my fields out of the `Grid` class without losing that awesome linkage you have does.

I would have liked to structure my `Product` and `ProductCrafted` data as I wish in my form automatically like this:

```
FileUpload::make('product.image'),
TextInput::make('others_informations'),
TextInput::make('product.name'),
TextInput::make('others_informations'),
```

Everything works when they are in the `relationship()` function but I would like to be able to place my fields where I want them in the middle of the multitude of data I have.

Any solutions?
Thanks very much!

**Comment by antoinemy** [comment, marked as answer]
It works like this : 

```
Grid::make()
  ->relationship('product')
  ->schema([
      TextInput::make('name'),
  ]),

TextInput::make('others_informations'),

Grid::make()
  ->relationship('product')
  ->schema([
      FileUpload::make('image')
          ->image(),
  ]),
```

Is it a good solution?

**Reply by danharrin** [reply]
Maybe swap Grid for Group do you don't need the columns

**Reply by Dhanidotini** [reply]
thats great, but im trying that in FileUpload and that cannot be deleted or if i wanna try remove image after crate post that say NOT NULL constraint failed, how to fix it?

fix: sorry, i just fixed that. just makesure the polymorph have nullable, that already explain here: [https://filamentphp.com/docs/4.x/forms/overview#saving-data-to-a-belongsto-or-morphto-relationship](url)
